tests/scripts/all.sh

```diff
@@ -547,6 +547,12 @@ if_build_succeeded tests/ssl-opt.sh -f Default
 msg "test: compat.sh RC4, DES & NULL (full config)" # ~ 2 min
 if_build_succeeded env OPENSSL_CMD="$OPENSSL_LEGACY" GNUTLS_CLI="$GNUTLS_LEGACY_CLI" GNUTLS_SERV="$GNUTLS_LEGACY_SERV" tests/compat.sh -e '3DES\|DES-CBC3' -f 'NULL\|DES\|RC4\|ARCFOUR'
 
+msg "build: cmake, full config + DEPRECATED_REMOVED, clang, C99"
+# No cleanup: tweak the configuration, keep the makefiles
+scripts/config.pl set MBEDTLS_DEPRECATED_WARNING
```

Comment:
Thanks for adding this (much needed) test. However, I would like to suggest that we enable more modules when doing the build test with `MBEDTLS_DEPRECATED_REMOVED`. Specially the modules that correspond to old crypto primitives that are disabled by default in Mbed TLS, where I would expect more deprecations (though in reality this might not be the case)...

Reply:
This runs in the `full` config minus `MBEDTLS_MEMORY_BACKTRACE`. Is there anything else to add?

Reply:
Cool, thanks for confirming. I think thats very complete

Reply:
Your question prompted me to check what else we didn't do in the full config, and one thing was a `-O` build, without which gcc and clang don't analyze the code very deeply. Doing a `-O` build caught an issue in `cmac.c`, so I added that.

Reply:
Thanks for checking :)